packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1880,7 +1886,9 @@ export class ContainerRuntime
 		await this.dataStores.waitIfPendingAlias(id);
 		const internalId = this.internalId(id);
 		const dataStoreContext = await this.dataStores.getDataStore(internalId, headerData);
-		const dataStoreChannel = await dataStoreContext.realize();
+		// Initialize snapshot details so that the packagePath is available which will be logged by GC in
+		// case of unexpected load of this data store.
+		await dataStoreContext.getInitialSnapshotDetails();
```

Comment:
From a quick glance it looks like some implementations don't do a Lazy Promise or anything so this will duplicate effort with the `realize` call below.

Reply:
Updated the implementations to use lazy promise.